### PR TITLE
Some important fixes and improvements

### PR DESCRIPTION
**Pull request by suutari-ai**
* Fix database table and colunm name generating.

 * Improve call interface of create_sanitized_dump.

 * Add support for PostGIS.

 * Add support for database URL forming for connection settings without
   username or password, or using a non-TCP communication such as unix
   socket.

See the commit messages for details.

### CODE REVIEW

File: sanitized_dump/utils/db.py [hunk 1]
```diff
@@ -31,8 +29,7 @@ def db_setting_to_db_string(db_settings):
             break
 
     if not database_type:
-        print('{} is not supported'.format(engine))  # noqa
-        return None
+        raise('Database type "{}" is not supported'.format(engine))
```
Comment (JsseL) [inline sanitized_dump/utils/db.py]: Just raise without specific type of an error?

File: sanitized_dump/utils/db.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-DATABASE_TYPES = ['mysql', 'postgres']
+DATABASE_TYPES = ['mysql', 'postgres', 'postgis']
```
Comment (JsseL) [inline sanitized_dump/utils/db.py]: Hmm.. but is `postgis` supported by the python-database-sanitizer?

**Comment by JsseL** [comment]
Good job figuring out the problems with m2m fields and the non "concrete" fields. This seems fine to me except the [one thing](https://github.com/andersinno/django-sanitized-dump/pull/14#discussion_r190544254) thing that I commented.

**Comment by JsseL** [comment]
Also, were you going to submit a PR to add the postgis support to the python-database-sanitizer package?

**Comment by frwickst** [comment]
Great work 👍Neither me nor Jesse has tried this on any proper project with the actual sanitizer so really nice to see that you was able to spot all the problems that was left in there!
image

**Comment by suutari-ai** [comment]
@JsseL Fixed the raise statement without an exception class. The PR to https://github.com/andersinno/python-database-sanitizer is coming too.

**Comment by suutari-ai** [comment]
The other PR: https://github.com/andersinno/python-database-sanitizer/pull/10

**Pull request merged**